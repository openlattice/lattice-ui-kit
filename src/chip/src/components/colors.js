import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
