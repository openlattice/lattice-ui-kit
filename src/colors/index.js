/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
