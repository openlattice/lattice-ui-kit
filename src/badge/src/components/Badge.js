// @flow
import styled from 'styled-components';
import * as Colors from '../../../colors';
import { getStyleVariation } from '../../../utils/StyleUtils';

const {
  GREENS,
  NEUTRALS,
  PURPLES,
  REDS,
  WHITE,
} = Colors;

const getBackgroundColor = getStyleVariation('mode', {
  added: GREENS[0],
  primary: PURPLES[2],
  removed: REDS[0],
  secondary: PURPLES[5],
}, NEUTRALS[6]);

const getFontColor = getStyleVariation('mode', {
  added: GREENS[3],
  primary: WHITE,
  removed: REDS[3],
  secondary: PURPLES[1],
}, NEUTRALS[0]);

const Badge = styled.span`
  border-radius: 32px;
  font-size: 12px;
  font-weight: normal;
  margin: 0 5px;
  padding: 3px 8px;
  background-color: ${getBackgroundColor};
  color: ${getFontColor};
`;

export default Badge;
