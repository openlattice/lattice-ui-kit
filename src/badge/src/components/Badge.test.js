import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Badge from './Badge';

import * as Colors from '../../../colors';

const { NEUTRAL, PURPLE } = Colors;

describe('Badge', () => {
  describe('snapshots', () => {

    test('should match snapshot', () => {
      const wrapper = mount(<Badge count="10" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('text should show max and not count', () => {
      const wrapper = mount(<Badge count="100" max="99" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('text should show count and not max', () => {
      const wrapper = mount(<Badge count="82" max="99" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('default max value should be displayed', () => {
      const wrapper = mount(<Badge count="8000" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('modes', () => {

    test('default', () => {
      const wrapper = mount(<Badge count="10" />);
      expect(wrapper.find(Badge)).toHaveStyleRule('color', NEUTRAL.N700);
      expect(wrapper.find(Badge)).toHaveStyleRule('background-color', NEUTRAL.N100);
    });

    test('primary', () => {
      const wrapper = mount(<Badge mode="primary" count="10" />);
      expect(wrapper.find(Badge)).toHaveStyleRule('color', 'white');
      expect(wrapper.find(Badge)).toHaveStyleRule('background-color', PURPLE.P300);
    });

    test('secondary', () => {
      const wrapper = mount(<Badge mode="secondary" count="10" />);
      expect(wrapper.find(Badge)).toHaveStyleRule('color', PURPLE.P300);
      expect(wrapper.find(Badge)).toHaveStyleRule('background-color', PURPLE.P100);
    });

  });
});
