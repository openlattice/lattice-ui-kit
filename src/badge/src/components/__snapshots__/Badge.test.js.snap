// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshots should match snapshot 1`] = `
.c0 {
  background-color: #f0f0f7;
  border-radius: 32px;
  color: #555e6f;
  font-size: 12px;
  font-weight: normal;
  margin: 0 5px;
  padding: 3px 8px;
}

<Badge
  max={0}
  mode=""
>
  <Badge__BadgeWrapper
    mode=""
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Badge__BadgeWrapper-sc-1kfg7sp-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:",
              [Function],
              ";border-radius:32px;color:",
              [Function],
              ";font-size:12px;font-weight:normal;margin:0 5px;padding:3px 8px;",
            ],
          },
          "displayName": "Badge__BadgeWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Badge__BadgeWrapper-sc-1kfg7sp-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      mode=""
    >
      <span
        className="c0"
        mode=""
      />
    </StyledComponent>
  </Badge__BadgeWrapper>
</Badge>
`;
